pyiceberg/table/__init__.py

```diff
@@ -988,11 +988,15 @@ def commit_transaction(self) -> Table:
         Returns:
             The table with the updates applied.
         """
-        self._requirements = (AssertCreate(),)
-        self._table._do_commit(  # pylint: disable=W0212
-            updates=self._updates,
-            requirements=self._requirements,
-        )
+        if len(self._updates) > 0:
+            self._table._do_commit(  # pylint: disable=W0212
+                updates=self._updates,
+                requirements=(AssertCreate(),),
+            )
+
+        self._updates = ()
+        self._requirements = ()
```

Comment:
We could refactor it like:
```suggestion
        super().commit_transaction()
```
But it doesn't actually commit, but just clear the `self_requirements`, which is also a bit confusing to me. For me, it makes less sense to commit a CreateTableTransaction twice in the first place.